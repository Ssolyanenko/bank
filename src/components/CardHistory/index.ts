export * from './History';
export * from './Pagination';
export * from './CardHistory';
