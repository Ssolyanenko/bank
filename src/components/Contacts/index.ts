export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
