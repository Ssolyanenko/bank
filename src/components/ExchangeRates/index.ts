export * from './ExchangeRates';
export * from './Converter';
export * from './Rates';
