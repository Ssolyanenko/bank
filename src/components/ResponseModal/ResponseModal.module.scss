@import 'src/styles/variables.module.scss';

.modalText {
  align-self: center;
  padding: 2.8vh 0 3.5vh;
  color: $grayDark_6;
}

.modalIcon {
  align-self: center;
}

.modalCloseButton {
  align-self: flex-end;
}

.modalFakeIcon {
  height: 100px;
}
