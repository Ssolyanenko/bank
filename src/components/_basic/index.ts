export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
