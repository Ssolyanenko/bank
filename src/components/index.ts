export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
