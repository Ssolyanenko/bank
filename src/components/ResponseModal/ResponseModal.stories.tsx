import React, { useEffect, useState } from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import { ResponseModal } from 'components/ResponseModal';
import { PrimaryButton } from 'components/_basic';
import { ERROR_MESSAGE } from 'constants/errors';
import { ERROR, SUCCESS } from 'constants/text';

export default {
  title: 'Basic/Modals/ResponseModal',
  component: ResponseModal,
  argTypes: {
    isOpened: {
      defaultValue: false,
      type: 'boolean',
      description: 'Shows either the modal is open or closed',
    },
    content: {
      defaultValue: ERROR_MESSAGE.somethingWentWrong,
      type: 'string',
      description: 'Text of from back-end',
    },
    status: {
      defaultValue: ERROR,
      type: 'string',
      description: 'Status of the card order. It is either success or error',
      options: [ERROR, SUCCESS],
      control: { type: 'select' },
    },
  },
  parameters: {
    docs: {
      description: {
        component:
          'Modal component that demonstrates different responses to POST request. Those include either error message or success message. Path: components/ModalOrderCard',
      },
    },
  },
} as ComponentMeta<typeof ResponseModal>;

const Template: ComponentStory<typeof ResponseModal> = ({ content, status, isOpened }) => {
  const [isModalOpen, setIsModalOpen] = useState(isOpened);

  useEffect(() => {
    setIsModalOpen(isOpened);
  }, [isOpened]);

  return (
    <>
      <PrimaryButton onClick={(): void => setIsModalOpen(true)}>Open Modal</PrimaryButton>
      <ResponseModal
        isOpened={isModalOpen}
        modalCloseHandler={(): void => setIsModalOpen(false)}
        content={content}
        status={status}
      />
    </>
  );
};

export const Default = Template.bind({});
