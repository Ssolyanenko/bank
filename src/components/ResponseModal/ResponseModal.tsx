import { FC, ReactElement } from 'react';

import { ModalConfirmation, CloseButton } from 'components/_basic';
import { Size } from 'interfaces/common/componentsSettings';
import { ICON_VALUES } from 'constants/modalWindow';
import { Statuses } from 'interfaces/statuses';
import classes from './ResponseModal.module.scss';

interface Props {
  isOpened: boolean;
  modalCloseHandler(): void;
  content: string | JSX.Element;
  status: Statuses;
}

export const ResponseModal: FC<Props> = ({ isOpened, modalCloseHandler, content, status }): ReactElement => (
  <ModalConfirmation isOpened={isOpened} onClose={modalCloseHandler} maxWidth="600px">
    <div className={classes.modalCloseButton}>
      <CloseButton size={Size.MEDIUM} onClick={modalCloseHandler} />
    </div>
    <div className={classes.modalIcon}>{ICON_VALUES[status]}</div>
    <p className={classes.modalText}>{content}</p>
  </ModalConfirmation>
);
