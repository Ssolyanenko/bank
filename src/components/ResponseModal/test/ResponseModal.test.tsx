import React from 'react';
import { shallow } from 'enzyme';

import { ResponseModal } from 'components';
import { ERROR, SUCCESS } from 'constants/text';
import { ERROR_MESSAGE } from 'constants/errors';
import { Default } from 'components/ResponseModal/ResponseModal.stories';

describe('<ResponseModal />', (): void => {
  test('should make shapshot of default modal', (): void => {
    const props = {
      isOpened: true,
      modalCloseHandler: jest.fn(),
      content: ERROR_MESSAGE.somethingWentWrong,
      status: ERROR as 'error',
    };

    const wrapper = shallow(<ResponseModal {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should make a snapshot of error modal', (): void => {
    const props = {
      isOpened: true,
      modalCloseHandler: jest.fn(),
      content: 'Invalid phone number. Try again.',
      status: ERROR as 'error',
    };

    const wrapper = shallow(<ResponseModal {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should make a snapshot of success modal', (): void => {
    const props = {
      isOpened: true,
      modalCloseHandler: jest.fn(),
      content: 'Your card order is received.',
      status: SUCCESS as 'success',
    };

    const wrapper = shallow(<ResponseModal {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should render ResponseModal from storybook', (): void => {
    const props = {
      isOpened: true,
      modalCloseHandler: jest.fn(),
      content: 'Your card order is received.',
      status: SUCCESS as 'success',
    };

    const wrapper = shallow(<Default {...props} />);

    expect(wrapper.find(ResponseModal).length).toBe(1);
  });
});
