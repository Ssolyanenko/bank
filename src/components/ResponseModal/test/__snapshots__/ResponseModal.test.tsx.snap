// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponseModal /> should make a snapshot of error modal 1`] = `
<ModalConfirmation
  isOpened={true}
  maxWidth="600px"
  onClose={[MockFunction]}
>
  <div
    className="modalCloseButton"
  >
    <CloseButton
      onClick={[MockFunction]}
      size="medium"
    />
  </div>
  <div
    className="modalIcon"
  >
    <ExclamationMarkIcon />
  </div>
  <p
    className="modalText"
  >
    Invalid phone number. Try again.
  </p>
</ModalConfirmation>
`;

exports[`<ResponseModal /> should make a snapshot of success modal 1`] = `
<ModalConfirmation
  isOpened={true}
  maxWidth="600px"
  onClose={[MockFunction]}
>
  <div
    className="modalCloseButton"
  >
    <CloseButton
      onClick={[MockFunction]}
      size="medium"
    />
  </div>
  <div
    className="modalIcon"
  >
    <CheckmarkIcon />
  </div>
  <p
    className="modalText"
  >
    Your card order is received.
  </p>
</ModalConfirmation>
`;

exports[`<ResponseModal /> should make shapshot of default modal 1`] = `
<ModalConfirmation
  isOpened={true}
  maxWidth="600px"
  onClose={[MockFunction]}
>
  <div
    className="modalCloseButton"
  >
    <CloseButton
      onClick={[MockFunction]}
      size="medium"
    />
  </div>
  <div
    className="modalIcon"
  >
    <ExclamationMarkIcon />
  </div>
  <p
    className="modalText"
  >
    Something went wrong. Please try again.
  </p>
</ModalConfirmation>
`;
